src/components/toggle/toggle.ts

```diff
@@ -0,0 +1,332 @@
+import {
+  AfterContentInit,
+  Component,
+  ContentChildren,
+  Directive,
+  EventEmitter,
+  HostBinding,
+  HostListener,
+  Input,
+  OnInit,
+  Optional,
+  Output,
+  QueryList,
+  ViewEncapsulation,
+  forwardRef
+} from '@angular/core';
+import {MdToggleDispatcher} from './toggle_dispatcher';
+
+
+export {MdToggleDispatcher} from './toggle_dispatcher';
+
+
+
+var _uniqueIdCounter = 0;
+
+/** A simple change event emitted by either MdToggle or MdToggleGroup. */
+export class MdToggleChange {
+  source: MdToggle;
+  value: any;
+}
+
+/** Exclusive selection toggle group. */
+@Directive({
+  selector: 'md-toggle-group:not([multiple])',
+  host: {
+    'role': 'radiogroup',
+  },
+})
+export class MdToggleGroup implements AfterContentInit {
+  /** The value for the toggle group. Should match currently selected toggle. */
+  private _value: any = null;
+
+  /** The HTML name attribute applied to toggles in this group. */
+  private _name: string = null;
+
+  /** Disables all toggles in the group. */
+  private _disabled: boolean = null;
+
+  /** The currently selected toggle, should match the value. */
+  private _selected: MdToggle = null;
+
+  /** Event emitted when the group's value changes. */
+  @Output()
+  change: EventEmitter<MdToggleChange> = new EventEmitter<MdToggleChange>();
+
+  /** Child toggle buttons. */
+  @ContentChildren(forwardRef(() => MdToggle))
+  private _toggles: QueryList<MdToggle> = null;
+
+  /**
+   * Initializes the names once content children are available.
+   * This allows us to propagate relevant attributes to associated toggles.
+   */
+  ngAfterContentInit() {
+    if (this.name == null) {
+      this.name = `md-toggle-group-${_uniqueIdCounter++}`;
+    } else {
+      this._updateChildToggleNames();
+    }
+  }
+
+  @Input()
+  get name(): string {
+    return this._name;
+  }
+
+  set name(value: string) {
+    this._name = value;
+
+    this._updateChildToggleNames();
+  }
+
+  /** Propagate name attribute to toggles. */
+  private _updateChildToggleNames(): void {
+    if (this._toggles != null) {
+      this._toggles.forEach((toggle) => {
+        toggle.name = this._name;
+      });
+    }
+  }
+
+  @Input()
+  get disabled(): boolean {
+    return this._disabled;
+  }
+
+  set disabled(value) {
+    this._disabled = (value != null && value !== false) ? true : null;
+  }
+
+  @Input()
+  get value(): any {
+    return this._value;
+  }
+
+  set value(newValue: any) {
+    if (this._value != newValue) {
+      this._value = newValue;
+
+      this._updateSelectedToggleFromValue();
+      this._emitChangeEvent();
+    }
+  }
+
+  private _updateSelectedToggleFromValue(): void {
```

Comment:
I'd like to find a way to extract this (and similar methods) into something resuable from both this component and radio-button, since they're effectively identical. Already the `radio.ts` version of this method has changed, so they're definitely get out of sync if they duplicate logic.